### PR TITLE
[table, test-commons] chore: avoid deprecated initMouseEvent API

### LINKED ISSUE #5173: HACKHACK: tests using deprecated initMouseEvent() method
test-commons and the table package test harness use the [deprecated `MouseEvent.initMouseEvent()` method](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/initMouseEvent). we should migrate to using the [`MouseEvent()` constructor](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/MouseEvent) instead.

### PR DESCRIPTION
**Pull request by SoulPancake**
#### Checklist

- [ ] Includes tests
- [ ] Update documentation

#### Changes proposed in this pull request:

Updated the deprecated MouseEvent.initMouseEvent() to the new MouseEvent() constructor and passed the same arguments and correct defaults according to the new documentation.

#### Reviewers should focus on:

Proper corresponding arguments for the functions as provided in the documentation.

### CODE REVIEW

**Comment by adidahiya** [comment]
@Anuragkillswitch please make sure to fix lint (`yarn lint-fix` in a package) and formatting (`yarn format` at the repo root). Let me know when the build passes and I'll review this PR.

**Comment by Anuragkillswitch** [comment]
@adidahiya I ran the yarn lint-fix in the packages but it didn't give me any errors locally, can you suggest what to do ?

**Comment by SoulPancake** [comment]
@adidahiya I have formatted and fixed the lint and the checks have passed, can you take a look? Thanks!

File: packages/test-commons/src/utils.ts [hunk 1]
```diff
@@ -101,10 +101,8 @@ function detectBrowser() {
 // see http://stackoverflow.com/questions/16802795/click-not-working-in-mocha-phantomjs-on-certain-elements
 // tl;dr PhantomJS sucks so we have to manually create click events
 export function createMouseEvent(eventType = "click", clientX = 0, clientY = 0) {
-    const event = document.createEvent("MouseEvent");
-
-    // https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/detail
```
Comment (adidahiya) [inline packages/test-commons/src/utils.ts:106]: why remove this comment?

File: packages/test-commons/src/utils.ts [hunk 2]
```diff
@@ -101,10 +101,8 @@ function detectBrowser() {
...
     let detailArg = 0;
```
Comment (adidahiya) [inline packages/test-commons/src/utils.ts:105]:
```suggestion
    // see https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/detail
    let detailArg = 0;
```

File: packages/test-commons/src/utils.ts [hunk 3]
```diff
@@ -116,25 +114,20 @@ export function createMouseEvent(eventType = "click", clientX = 0, clientY = 0)
...
+        cancelable: true,
+        clientX,
+        clientY,
+        detail: detailArg,
+        view: window,
+    });
+    // https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/detail
+
     // HACKHACK: see https://github.com/palantir/blueprint/issues/5173
     // eslint-disable-next-line deprecation/deprecation
```
Comment (adidahiya) [inline packages/test-commons/src/utils.ts]: wrong location for this MDN link, also we should remove the HACKHACK and eslint comment flag

```suggestion

```

**Review by adidahiya — CHANGES_REQUESTED** [review]
couple minor code style things, otherwise lgtm

**Review by adidahiya — APPROVED** [review]

**Pull request merged**